src/mcts/search.cc

```diff
@@ -518,16 +521,27 @@ bool SearchWorker::IsSearchActive() const {
 // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 void SearchWorker::InitializeIteration(
     std::unique_ptr<NetworkComputation> computation) {
-  nodes_to_process_.clear();
+  // If last iteration was out of order eval, continue gathering batch.
+  if (out_of_order_eval_) {
+    // If NN eval was already processed on the last iteration, remove it.
+    if (nodes_to_process_.back().nn_queried) computation_->PopInput();
+    // Was already processed on the last iteration.
+    nodes_to_process_.pop_back();
+    out_of_order_eval_ = false;
+    return;
+  }
+
+  // It was not an out-of-order eval, batch is fresh.
   computation_ = std::make_unique<CachingComputation>(std::move(computation),
                                                       search_->cache_);
+  nodes_to_process_.clear();
+  collisions_found_ = 0;
 }
 
 // 2. Gather minibatch.
 // ~~~~~~~~~~~~~~~~~~~~
 void SearchWorker::GatherMinibatch() {
-  int nodes_found = 0;
-  int collisions_found = 0;
+  int nodes_found = nodes_to_process_.size() - collisions_found_;
```

Comment:
Is this correct? collisions can also be due to collisions with other threads batches, if we subtract those from our batch, our batch might get smaller and smaller. Maybe a reason for the decreasing nps some observed